src/main/java/io/mathdojo/Question.java

```diff
@@ -1,34 +1,56 @@
 package io.mathdojo;
 
+import java.util.Arrays;
+import java.util.Map;
+
 import org.springframework.data.annotation.Id;
 import org.springframework.data.mongodb.core.mapping.Document;
 
 
 @Document(collection = "questions")
-public class Question {
-	
+public class Question {	
+	public static Question EMPTY_DATABASE =  new Question("the-database-is-empty", null, null, null, 0, null, null, null, null, null);
+	//id must be kebab-case
 	@Id
+	private String id;
     private String questionTitle;
     private String questionBody;
     private String sampleAnswer;
-    private String[] hints;
-    private int successRate;
+    private Integer successRate;
     private String difficulty;
-    private boolean solved;
+    private String[] hints;
+    private String parentTopicTitle;
+    private String[] questionAnswerOptions;
+    private String answer;
 
     public Question() {
     }
 
-	public Question(String questionTitle, String questionBody, String sampleAnswer, String[] hints, int successRate,
-			String difficulty, boolean solved) {
+	public Question(String id, String questionTitle, String questionBody, String sampleAnswer, Integer successRate,

```

Comment:
We've got some inconsistencies between the names of our objects here and the swagger docs:
https://github.com/math-dojo/question-service/blob/aa2ec892efc3a5645986d46bfa7028a63612b1c9/openapi.yaml#L320

Would you like to resolve them or shall I? @aakano684

Reply:
I reckon this could be a problem when the data is serialised to JSON the WebUI tries to make a call for this data. We could use the API could transpose these to fields the UI is expecting (which is fine) but the swagger should match up.

Reply:
I'll change to match the spec